src/assets.zig

```diff
@@ -402,30 +451,81 @@ pub fn loadWorldAssets(assetFolder: []const u8, blockPalette: *Palette, biomePal
 		_ = main.models.registerModel(entry.key_ptr.*, entry.value_ptr.*);
 	}
 
-	// blocks:
 	blocks_zig.meshes.registerBlockBreakingAnimation(assetFolder);
-	for(blockPalette.palette.items) |id| {
-		const nullValue = blocks.get(id);
-		var zon: ZonElement = undefined;
-		if(nullValue) |value| {
-			zon = value;
-		} else {
-			std.log.err("Missing block: {s}. Replacing it with default block.", .{id});
-			zon = .null;
-		}
-		try registerBlock(assetFolder, id, zon);
+
+	// Blocks:
+	// First blocks from the palette to enforce ID values.
+	for(blockPalette.palette.items) |stringId| {
+		try registerBlock(assetFolder, stringId, blocks.get(stringId) orelse .null);
 	}
+
+	// Then all the blocks that were missing in palette but are present in the game.
 	var iterator = blocks.iterator();
 	while(iterator.next()) |entry| {
-		if(blocks_zig.hasRegistered(entry.key_ptr.*)) continue;
-		try registerBlock(assetFolder, entry.key_ptr.*, entry.value_ptr.*);
-		blockPalette.add(entry.key_ptr.*);
+		const stringId = entry.key_ptr.*;
+		const zon = entry.value_ptr.*;
+
+		if(blocks_zig.hasRegistered(stringId)) continue;
+
+		try registerBlock(assetFolder, stringId, zon);
+		blockPalette.add(stringId);
+	}
+
+	// Items:
+	// First from the palette to enforce ID values.
+	for(itemPalette.palette.items) |stringId| {
+		std.debug.assert(!items_zig.hasRegistered(stringId));
+
+		// Some items are created automatically from blocks.
+		if(blocks.get(stringId)) |zon| {
+			if(!zon.get(bool, "hasItem", true)) continue;
+			try registerItem(assetFolder, stringId, zon.getChild("item"));
+			if(items.get(stringId)) |_| {
```

Comment:
If you don't need the capture then you can also use `!= null`

Reply:
```suggestion
			if(items.get(stringId) != null) {
```